src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -637,19 +637,20 @@ public override void DidReceiveData (NSUrlSession session, NSUrlSessionDataTask
 			public override void DidCompleteWithError (NSUrlSession session, NSUrlSessionTask task, NSError error)
 			{
 				var inflight = GetInflightData (task);
+				var serverError = task.Error;
```

Comment:
Minor: if you changed this to:
```
error = error ?? task.Error;
```
Then you don't need any of the other changes.

Reply:
Unfortunately.. is a backport :/ I wish you have been in the first review :(